gateway/gateway.go

```diff
@@ -479,6 +470,29 @@ func (g *gateway) DoCommand(ctx context.Context, cmd map[string]interface{}) (ma
 	if _, ok := cmd["validate"]; ok {
 		return map[string]interface{}{"validate": 1}, nil
 	}
+	if deviceReq, ok := cmd["return_devices"]; ok {
+		resp := map[string]interface{}{}
+		deviceStr, _ := deviceReq.(string)
+		// look for a specific device if requested. If the device name is not a string we just return all devices.
+		if deviceStr != "" {
+			// check if the device has joined the network.
+			resp[deviceStr], ok = g.devices[deviceStr]
+			if ok {
+				resp["specific device currently joined"] = deviceStr
```

Comment:
why do we set this only to inevitably not use this on line 483?

Reply:
I missed that this part of the DoCommand was no longer functional. I ended up only using/testing the all devices path.

Reply:
removed the deviceStr logic